src/main/java/frc/robot/subsystems/vision/Vision.java

```
@@ -0,0 +1,95 @@
+          : null;
+      photonPoseEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
+      // }
+    } catch (Exception e) {
+      DriverStation.reportError("Failed to load AprilTagFieldLayout", e.getStackTrace());
+      // photonPoseEstimator = null;
+    }
+
+    this.photonPoseEstimator = photonPoseEstimator;
+  }
+
+  @Override
+  public void run() {
+    // Get AprilTag data and updating the pose estimator
+    try {
+      if (photonPoseEstimator != null && photonCamera != null) {
+        var photonResults = photonCamera.getLatestResult(); //Gets the latest camera results
+        SmartDashboard.putString(photonCamera.getName(),
+            photonCamera.getLatestResult().getMultiTagResult().estimatedPose.toString());
+        if (photonResults.hasTargets() && (photonResults.targets.size() > 1
+            || photonResults.targets.get(0).getPoseAmbiguity() < VisionConstants.APRILTAG_AMBIGUITY_THRESHOLD)) {
+          //Updates the pose estimator
+          photonPoseEstimator.update(photonResults).ifPresent(estimatedRobotPose -> {
+            var estimatedPose = estimatedRobotPose.estimatedPose;
+            // SmartDashboard.putNumber("estimatedPoseX", estimatedPose.toPose2d().getX());
+            // SmartDashboard.putNumber("estimatedPoseY", estimatedPose.toPose2d().getY());
+            /** 
+            * If present then makes sure the measurement is on the field and
+            * sets the atomic estimated pose to the current estimated pose
+            * from PhotonPoseEstimator
+            */
+            if (estimatedPose.getX() > 0.0 && estimatedPose.getX() <= FieldConstants.length
+                && estimatedPose.getY() > 0.0
+                && estimatedPose.getY() <= FieldConstants.width) {
+              atomicEstimatedRobotPose.set(estimatedRobotPose);
+            }
+          });
+        }
+      }
+    } catch (Exception e) {
+      // TODO: handle exception
```

Why catch all exceptions if you aren't going to handle them? better to let it crash if you have an exception and don't know what to do with it

---

Its so we can see which camera is crashing